client/components/settings/local_settings_modal_controller.js

```diff
@@ -0,0 +1,41 @@
+class LocalSettingsModalController {
+  constructor (localSettings) {
+    Object.assign(this, { localSettings })
+    this.settings = {}
+  }
+
+  $onInit () {
+    this.loadSettings()
+  }
+
+  loadSettings () {
+    this.settings = this.localSettings.getFromLocalStorage()
+    this.settingsData = {}
+    // settingsData makes it much easier to properly display the settings in the modal itself
+    Object.keys(this.settings).forEach(settingsKey => {
```

Comment:
Using `Object.entries` will remove the need to use `this.ettings[settingsKey]` each time you need the value.